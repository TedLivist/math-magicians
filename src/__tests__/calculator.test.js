import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

describe('Calculator snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Calculator funuctionality', () => {
  test('Displays number 1 when button', () => {
    render(<Calculator />);

    const btnOne = screen.getByText('1');
    const input = screen.getByPlaceholderText('0');
    fireEvent.click(btnOne);

    expect(input.value).toBe('1');
  });

  test('Displays second number after operator', () => {
    render(<Calculator />);

    const btnOne = screen.getByText('1');
    const btnTwo = screen.getByText('2');
    const btnPlus = screen.getByText('+');
    const input = screen.getByPlaceholderText('0');
    fireEvent.click(btnOne);
    fireEvent.click(btnPlus);
    fireEvent.click(btnTwo);

    expect(input.value).toBe('2');
  });
});
