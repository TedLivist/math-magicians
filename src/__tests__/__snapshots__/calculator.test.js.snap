// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator-root"
>
  <h1>
    Let's do some math!
  </h1>
  <div
    className="container"
  >
    <input
      className="input-box"
      placeholder="0"
      readOnly={true}
    />
    <div
      className="sub-container"
    >
      <button
        className="ac button-width left-button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="plus-minus button-width left-button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="percent button-width left-button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="divide button-width right-button"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="sub-container"
    >
      <button
        className="seven button-width left-button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="eight button-width left-button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="nine button-width left-button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="times button-width right-button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="sub-container"
    >
      <button
        className="four button-width left-button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="five button-width left-button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="six button-width left-button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="minus button-width right-button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="sub-container"
    >
      <button
        className="four button-width left-button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="five button-width left-button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="six button-width left-button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="minus button-width right-button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="last-container"
    >
      <button
        className="zero left-button"
        id="zero-button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="dot button-width left-button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="equal button-width right-button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
